indexd/fence_client.py

```diff
@@ -40,6 +40,8 @@ def get_signed_url_for_object(self, object_id, access_id):
             raise IndexNoRecordFound(
                 "No document with id:{} with access_id:{}".format(object_id, access_id)
             )
+        if req.status_code == 401:
+            raise AuthzError("Unauthorized: Access denied due to invalid credentials.")
         elif req.status_code != 200:
             raise UnexpectedError(req.text)
```

Comment:
add a log here before raising 500, with status code and details?
unless you're just deleting this sign URL creation soon